### ISSUE TITLE
2.8.2 regression: backgound powerBI scanning not working

### ISSUE THREAD

**Issue by zenzeinet** [DaxStudio/DaxStudio#155]
Hi,
I could connect using version 2.8.1 to local powerBI models, but version 2.8.2 stays stuck at loading without geting any results. Seems related to the new feature changelog of 
"Scanning for Power BI instances in connection dialog moved to a background thread". 
Tell me if I can help anyhow.

DAX Studio Checker of 2.8.1 (working well) below. Windows Locale OS in Spanish. Installation folder different from usual - i.e., not installed in c:\program files (x86)\DaxStudio, but in another folder. Not sure if any other unusual config that could be affecting the results.

**Comment by dgosbell** [comment]
We don't really need the output from DAXStudio.Checker in this situation - we mainly use that if anyone has install or startup issues so I've removed that content to make this thread easier to read (plus it's the output from the working version so I would expect to see any issues). The installation folder should not matter, I'm often running preview versions from different folders. 

I tested this feature for a number of weeks with multiple versions of Power BI and SSDT and with multiple files open at a time. The only time I've seen it get stuck at "loading" is when you have a dialog box of some sort open (like file save as) or the Get Data window open. However in DAX Studio 2.8.1 this would freeze the _entire_ application, not just the combo box in the connection window.

How many Power BI files did you test this with - does it happen on 1 particular file or on many?
Did you have any dialog boxes open in Power BI? 

If you wanted to test 2.8.2 again and it's still having this issue would you mind trying to capture some debug logs? You just need to start up DAX Studio using [these instructions](https://daxstudio.org/documentation/troubleshooting/logging-settings/), reproduce the issue, then close down DAX Studio and upload the log to this thread. Feel free to look through the log first if you like, there should not be any personal information in there, but there may be some file paths (which you could edit to mask out bits of the path if you'd prefer not to have that posted)

**Comment by dgosbell** [comment]
Could run the following in either Powershell ISE or a Powershell window after opening a report with a data model in Power BI Desktop?
```
$process = "MSMDSRV"
get-process -name $process | % {get-nettcpconnection -OwningProcess $_.id} | select-object @{Name="Process"; Expression = {$Process}},LocalPort, RemotePort
$process = "PBIDesktop"
get-process -name $process | % {get-nettcpconnection -OwningProcess $_.id} | select-object @{Name="Process"; Expression = {$Process}},LocalPort, RemotePort
``` 
Can you then open DAX Studio with the debug logging on(see [https://daxstudio.org/documentation/troubleshooting/logging-settings/]()) In the debug log file there should be one or more lines like the following (depending on how many PowerBI Desktop instances you have open)
```
2019-03-04 20:49:09.021 -08:00 [Debug] "PowerBIHelper" "Refresh" PowerBI found on port: 51130
```
Could you run the following, replacing XXXX with the port number that DAX Studio found?
```
Get-Process -Id (Get-NetTCPConnection -LocalPort XXXX ).OwningProcess | fl * 
```
If DAX Studio is not getting the port number correct this may give me some hints as to why.

If you can either paste the output of the above powershell commands back here or attach it in a txt file that would help a lot.

**Comment by dgosbell** [comment]
I just released v2.8.3 which has a few improvements to the Power BI scanning, one in particular that should improve the scanning speed and another which will improve the error logging. Can you please try this and see if it helps with this issue?

**Comment by dgosbell** [comment]
Assuming that 2.8.3 resolved this issue
